skorch/utils.py

```diff
@@ -274,3 +278,18 @@ def noop(*args, **kwargs):
     target extractor.
     """
     pass
+
+
+@contextmanager
+def _with_file_like(f, mode):
+    new_fd = False
+    if (
+        isinstance(f, str) or
+        sys.version_info[0] == 2 and isinstance(f, unicode) or
+        sys.version_info[0] == 3 and isinstance(f, pathlib.Path)
+    ):
+        new_fd = True
+        f = open(f, mode)
+    yield f
+    if new_fd:
+        f.close()
```

Comment:
Do we need a `try ... finally` here in case we open the file but something goes wrong?

Reply:
If there is an error opening the file, I think the caller should handle the exception.